tests/Spec/SubspecTest.php

```diff
@@ -0,0 +1,63 @@
+<?php
+namespace Aura\Filter\Spec;
+
+use Aura\Filter\FilterFactory;
+use Yoast\PHPUnitPolyfills\TestCases\TestCase;
+
+class SubspecTest extends TestCase
+{
+    protected $filter;
+
+    protected function set_up()
+    {
+        $filter_factory = new FilterFactory();
+        $this->filter = $filter_factory->newSubjectFilter();
+    }
+
+    public function testMultidimensionalSupport()
+    {
+        $data = [
+            'id' => 'asd',
+            'user' => [
+                'name' => 'Foo',
+                'age' => 'asd',
+            ],
+            // 'url' => 'http://example.com'
+        ];
+
+        $this->filter->validate('id')->is('int');
+        $this->filter->validate('url')->is('url');
+
+        $user_spec = $this->filter->subfilter('user'); // add a "SubSpec"
```

Comment:
We could probably rename subfilter to subFilter .

Reply:
I created PR https://github.com/harikt/Aura.Filter/pull/2